### PR TITLE
ISSUE-1204 Documentation contact indices settings migration

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

File: upgrade-instructions.md [hunk 1]
```diff
@@ -12,23 +12,325 @@ software documentation. Do not follow this guide blindly!
...
+      },
+      "email": {
+        "type": "text",
+        "analyzer": "email_ngram_filter_analyzer",
+        "search_analyzer": "rebuilt_keyword"
+      },
+      "firstname": {
+        "type": "text",
+        "analyzer": "name_edge_ngram_filter_analyzer",
+        "search_analyzer": "standard"
```
Comment (chibenwa) [inline upgrade-instructions.md:104]: Is this intended to keep standard here?

Comment (quantranhong1999) [reply upgrade-instructions.md:104]: I think so. What is the expectation to not keep search_analyzer as standard?

Comment (chibenwa) [reply upgrade-instructions.md:104]: Having a `standard` search analyzer means filtering.

Example

GIVEN `Andre` 
WHEN I type `and` in the search bar
THEN the standard analyzer thinks it is an english stopword and filters out the token thus rendering the search useless.

Treating the search as a keyword (tokenized the right way) would avoid such pitfalls.

I'm interested by the test of andre/and BTW to see if I am right.

Comment (HoussemNasri) [reply upgrade-instructions.md:104]: Relevant documentation: https://opensearch.org/docs/latest/analyzers/

Comment (HoussemNasri) [reply upgrade-instructions.md:104]: `Keyword` seem like the analyzer we want to use here, given that we don't want to split the emails before indexing.

Comment (quantranhong1999) [reply upgrade-instructions.md:104]: > I'm interested by the test of andre/and BTW to see if I am right.

I added a test. We should be good already.

Comment (quantranhong1999) [reply upgrade-instructions.md:104]: > Overall Iam sory I was convinced the standard analyzer was doing more work here.

Don't be sorry. It is always good to check though.

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**